### ISSUE TITLE
[V3.0.0][ANDROID] TypeError: null is not an object (evaluating 'this.nativeCommandsModule.refreshToken')

### ISSUE THREAD

**Issue by bduyng** [wix/react-native-notifications#440]
as the title said. I am using the newest version of the module on Android. but I got this error when calling `Notifications.registerRemoteNotifications`

The error is: `TypeError: null is not an object (evaluating 'this.nativeCommandsModule.refreshToken')`

My package.json
```
"react": "16.9.0",
"react-native": "0.61.2",
"react-native-notifications": "^3.0.0",
```

Please fix the problem! I really appreciate it!!!

**Comment by SMKH-PRO** [comment]
I am getting  undefined:
TypeError: Cannot read property 'refreshToken' of undefined


<img src="https://i.postimg.cc/bvzVbs8M/Screenshot-1579136405.png" />

**Comment by fgreinus** [comment]
> I am getting undefined:
> TypeError: Cannot read property 'refreshToken' of undefined
> 
> ![](https://camo.githubusercontent.com/6fb18897ed215d7b2857c2abf0994061b4fa392d/68747470733a2f2f692e706f7374696d672e63632f62767a566273384d2f53637265656e73686f742d313537393133363430352e706e67)

This usually means, that you have not properly linked the native module for the current platform OS.

**Comment by bduyng** [comment]
Confirmed that the problem is about linking

**Comment by JaioSkura** [comment]
I think that I have already well linked, but showing this error anyway. 
 "react": "16.9.0",
 "react-native": "0.61.5",
 "react-native-notifications": "3.1.1",

How is this issuel solved?

**Comment by ShaneMatthias** [comment]
@bduyng was right. It was a linking issue.

**Comment by JaioSkura** [comment]
yeah, I know. But how can  I solve it? 
Autolinking seems to be not working. I follow this steps, to link manually: https://github.com/wix/react-native-notifications/issues/447#issuecomment-576744535
But I continue having the same error in Android.

**Comment by ShaneMatthias** [comment]
Hmm.. Im not too sure. I forgot to add this part and after adding it, it worked for me.

In `MainApplication.java`
```
import com.wix.reactnativenotifications.RNNotificationsPackage;

protected List<ReactPackage> getPackages() {
  ...
  packages.add(new RNNotificationsPackage(MainApplication.this));
  return packages;
}
```

**Comment by JaioSkura** [comment]
Wow! thanks so much! I was adding that line, but not right. So stupid...

You save my day, making me see that concrete line carefully. Thanks a lot!

**Comment by lylest** [comment]
I am facing the same problem can you provide the full sample code of your solution  especial that linking part i am stuck here @JaioSkura

**Comment by JaioSkura** [comment]
Hi @lylest 
What react-native version are you using? and what react-native-notifications version? In my case, with :
"react-native": "0.61.5",
"react-native-notifications": "3.1.4"

you don't need to do anything special. This solution was for 3.1.1 but with the actual version the autolink feature works right. 

Let me know!

**Comment by lylest** [comment]
iIam using 
"react-native": "0.62.2",
 "react-native-notifications": "^3.1.4", "react-native-notifications": "^3.1.4",
it doesn't work or should I remove manual linking that i have set i am facing the same problem exactly as above screen shots

**Comment by JaioSkura** [comment]
Hi, I would try removing manual linking yes.

**Comment by lylest** [comment]
failed with same error as on the screen shot above

**Comment by JaioSkura** [comment]
I haven't try with RN 0.62, but is strange having a regresion in that sense. Do you have somewhere manually included the library? build.gradle, settings.gradle or mainAplication.java?
If so delete, clean , i would remove build folder too, and rebuild all.

**Comment by lylest** [comment]
let Cross check again and try to clean and build again

**Comment by seif1000** [comment]
I FOLLOWED THE DOCS STEP BY STEP BUT STILL HAVE THIS ISSUE I TRIED AUTO LINKING AND MANUAL LINKIG  BUT NONE OF THEM WORK ANY UPDATED SOLUTION FOR THIS ISSUE

**Comment by JaioSkura** [comment]
Witch versions of react-native and react-native-notifications are you using?

**Comment by microxoft** [comment]
I tried all these ideas, without success.

"react-native": "0.62.2",
"react-native-notifications": "^3.2.2",

Having the same error displayed on the screenshot. Tried cleaning folders and chaces, rebuilding, resyncing and no luck.

**Comment by lylest** [comment]
I actually failed to implement notifications for my app the documentations
are awful and most videos are outdated

**Comment by microxoft** [comment]
Fine, giving a try to https://github.com/zo0r/react-native-push-notification in that case.

Thanks,

**Comment by jkoutavas** [comment]
Manually linking react-native-notifications and adding this bit to `react-native.config.js` solved it for me on our react-native 0.61.0 app.

```
  dependencies: {
    'react-native-notifications': {
      platforms: {
        android: null, // disable Android platform, other platforms will still autolink if provided
      },
    },
  },
```

**Comment by Vanclief** [comment]
@ShaneMatthias solution works perfectly.


My MainApplication.java:
```
    protected List<ReactPackage> getPackages() {
      return Arrays.<ReactPackage>asList(
          new MainReactPackage(),
            ... 
          new RNNotificationsPackage(MainApplication.this)
      );
    }
```
